src/managers/resources/emg.rs

```diff
@@ -29,16 +81,82 @@ impl Resource for Emg {
 
 impl ResourceManager for Manager<Emg> {
     type ResourceType = Emg;
-    /// Handles all EMG-related tasks
+
     async fn handle_task(&mut self, channel_data: ManagerChannelData) -> Result<()> {
-        let (task, _, send_channel) =
+        let (task, task_data, send_channel) =
             parse_channel_data!(channel_data, Task, EmgData).map_err(|e: Error| e)?;
-        match task {
-            Task::UndefinedTask => todo!(),
-        }
+
+        let res = match task {
+            Task::UndefinedTask => {
+                warn!("Encountered an undefined task type");
+                Err(Error::msg("Encountered an undefined task type"))
+            }
+            Task::Idle => {
+                match adc::read_adc_channels(&[0, 1], &mut self.resource.cs_pin, &mut self.resource.spi) {
+                    Ok(value) => {
+                        info!("EMG ADC Channel 0,1 value: {:?}", value);
+                        match adc::process_data(value, self.resource.inner_threshold, self.resource.outer_threshold) {
+                            Ok(grip_state) => {
+                                info!("Grip state: {:?}", grip_state);
+                                if grip_state == 1 {
+                                    info!("Opening hand");
+                                    Ok("OPEN HAND".to_string())
+                                } else { // todo handle the case where grip_state is -1
+                                    info!("Closing hand");
+                                    Ok("CLOSE HAND".to_string())
+                                }
+                            }
+                            Err(e) => {
+                                error!("Failed to process EMG data: {:?}", e);
+                                Err(Error::msg("Failed to process EMG data: {}"))
+                            }
+                        }
+                    }
+                    Err(e) => {
+                        error!("Failed to read from EMG SPI: {:?}", e);
+                        Err(Error::msg("Failed to read SPI"))
+                    }
+                }
+            }
```

Comment:
Something lilke the following would be more readable compared to nested match statements. In general, if you wish to propagate errors in the Err case, you could use `?` for "flatter" code -- which is usually more readable
```suggestion
            Task::Idle => {
                let adc_values = adc::read_adc_channels(&[0, 1], &mut self.resource.cs_pin, &mut self.resource.spi)?;
                info!("EMG ADC Channel 0,1 value: {:?}", value);
                
                let grip_state = adc::process_data(value, self.resource.inner_threshold, self.resource.outer_threshold)?;
                info!("Grip state: {:?}", grip_state);
                
               if grip_state == 1 {
                   info!("Opening hand");
                   Ok("OPEN HAND".to_string())
              } else { // todo handle the case where grip_state is -1
                   info!("Closing hand");
                   Ok("CLOSE HAND".to_string())
              }
          }
```